contracts/price/ZNSCurvePricer.sol

```diff
@@ -159,83 +165,123 @@ contract ZNSCurvePricer is AAccessControlled, ARegistryWired, UUPSUpgradeable, I
      * @notice Sets the max price for domains. Validates the config with the new price.
      * Fires `MaxPriceSet` event.
      * Only domain owner can call this function.
-     * > `maxPrice` can be set to 0 along with `baseLength` or `minPrice` to make all domains free!
-     * @dev We are checking here for possible price spike at `maxLength` if the `maxPrice` values is NOT 0.
-     * In the case of 0 we do not validate, since setting it to 0 will make all subdomains free.
+     * > `maxPrice` can be set to 0 along with `baseLength` to make all domains free!
+     * > `maxPrice` cannot be 0 when:
+     *   - `maxLength` is 0;
+     *   - `baseLength` AND `curveMultiplier` are 0;
+     * @dev In the case of 0 we do not validate, since setting it to 0 will make all subdomains free.
+     * @param domainHash The domain hash to set the `maxPrice` for it
      * @param maxPrice The maximum price to set
      */
     function setMaxPrice(
         bytes32 domainHash,
         uint256 maxPrice
     ) external override onlyOwnerOrOperator(domainHash) {
         priceConfigs[domainHash].maxPrice = maxPrice;
-
-        if (maxPrice != 0) _validateConfig(domainHash);
-
         emit MaxPriceSet(domainHash, maxPrice);
     }
 
     /**
-     * @notice Sets the minimum price for domains. Validates the config with the new price.
-     * Fires `MinPriceSet` event.
-     * Only domain owner/operator can call this function.
-     * @param domainHash The domain hash to set the `minPrice` for
-     * @param minPrice The minimum price to set in $ZERO
+     * @notice Sets the multiplier for domains calculations
+     * to allow the hyperbolic price curve to be bent all the way to a straight line.
+     * Validates the config with the new multiplier in case where `baseLength` is 0 too.
+     * Fires `CurveMultiplier` event.
+     * Only domain owner can call this function.
+     * - If `curveMultiplier` = 1.000 - default. Makes a canonical hyperbola fucntion.
+     * - It can be "0", which makes all domain prices max.
+     * - If it is less than 1.000, then it pulls the bend towards the straight line.
+     * - If it is bigger than 1.000, then it makes bigger slope on the chart.
+     * @param domainHash The domain hash to set the price config for
+     * @param curveMultiplier Multiplier for bending the price function (graph)
      */
-    function setMinPrice(
+    function setCurveMultiplier(
         bytes32 domainHash,
-        uint256 minPrice
+        uint256 curveMultiplier
     ) external override onlyOwnerOrOperator(domainHash) {
-        priceConfigs[domainHash].minPrice = minPrice;
+        CurvePriceConfig memory config = priceConfigs[domainHash];
+        _validateCurveMultiplier(domainHash, curveMultiplier, config);
+        emit CurveMultiplierSet(domainHash, curveMultiplier);
+    }
 
-        _validateConfig(domainHash);
+    function _validateCurveMultiplier(
```

Comment:
call this `_validateSetCurveMultiplier()`.
A "validate" function can be `view`, this function is not view cause it actually sets the multiplier. confusing name